### PR TITLE
chore: add docker container healthcheck for dsp-api (INFRA-93)

### PR DESCRIPTION
**Pull request by SamuelBoerlin**
### Task Description/Number

Issue Number: [INFRA-93](https://linear.app/dasch/issue/INFRA-93/reboot-docker-container-if-dsp-api-is-crashed)

This change allows the container to detect when it is unhealthy and restart automatically in case the API is no longer responding or fatally unhealthy.

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [ ] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [x] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

[INFRA-93]: https://dasch.atlassian.net/browse/INFRA-93?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapi/scripts/healthcheck.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+json=$(curl -sS --max-time 10 'http://localhost:3339/health')
+if [ $? -ne 0 ]; then
+  echo "Health route is not responding"
+  exit 1
+fi
+
+# Check severity reported by health route
+severity=$(echo "$json" | jq -r .severity)
+if [ "$severity" != "non fatal" ]; then
+  echo "Heath route reports non non fatal severity"
```
Comment (irinaschubert) [inline webapi/scripts/healthcheck.sh:13]:
```suggestion
  echo "Health route reports other than a non fatal severity"
```

File: webapi/scripts/healthcheck.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Fetch health route status json
+json=$(curl -sS --max-time 10 'http://localhost:3339/health')
+if [ $? -ne 0 ]; then
+  echo "Health route is not responding"
+  exit 1
+fi
+
+# Check severity reported by health route
```
Comment (irinaschubert) [inline webapi/scripts/healthcheck.sh:10]:
```suggestion
# Check severity reported by health route, exit with 1 if severity is other than "non fatal"
```

**Review by irinaschubert — APPROVED** [review]
LGTM, just some suggestions to make things a bit clearer.

**Review by seakayone — APPROVED** [review]

File: webapi/scripts/healthcheck.sh (hunk 1 above)
Comment (SamuelBoerlin) [reply webapi/scripts/healthcheck.sh:13]: Yep that's better, thanks 👍

**Pull request merged**